pymodbus/client/sync.py

```diff
@@ -286,6 +294,35 @@ def _recv(self, size):
 
         return b"".join(data)
 
+    def _handle_abrupt_socket_close(self, size, data, duration):
+        """ Handle unexpected socket close by remote end
+
+        Intended to be invoked after determining that the remote end
+        has unexpectedly closed the connection, to clean up and handle
+        the situation appropriately.
+
+        :param size: The number of bytes that was attempted to read
+        :param data: The actual data returned
+        :param duration: Duration from the read was first attempted
+               until it was determined that the remote closed the
+               socket
+        :return: The more than zero bytes read from the remote end
+        :raises: ConnectionException If the remote end didn't send any
+                 data at all before closing the connection.
+        """
+        self.close()
+        readsize = (f"read of {size} bytes" if size
+                    else "unbounded read")
+        msg = (f'{self.__str__()}: Connection unexpectedly closed '
```

Comment:
fstrings are supported only in python3, this would essentially the PR valid for pymodbus 3.x. If you intend to support to python 2.x or want this to go in to pymodbus 2.5.0 please update accordingly. Otherwise please raise this against branch 3.0.0

Reply:
Thank you, good catch, this was a copy from the custom branch that Gapit uses, which is only 3.0. I don't have any thought of whether this should go into the 2.x or 3.x versions. Might as well update it to work for 2.x.